### PR TITLE
chore: Use cypress interceptors instead of wait to fix file csv flaky test

### PR DESCRIPTION
**Pull request by rajatagrawal**
This PR fixes the flaky test seen in FilePickerV2_CSV_spec.js.

The fix has been to replace wait with network interceptors.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const commonlocators = require("../../../../../locators/commonlocators.json");
 const dsl = require("../../../../../fixtures/filePickerTableDSL.json");
+import { ObjectsRegistry } from "../../../../../support/Objects/Registry";
+const locator = ObjectsRegistry.CommonLocators;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:4]: Import from ObjectsCore.
```
import {
locators,
} from "../../../../support/Objects/ObjectsCore";
```

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 2]
```diff
@@ -30,14 +32,17 @@ describe("File picker widget v2", () => {
     cy.get(
       `.t--property-control-dataformat ${commonlocators.helperText}`,
     ).contains(ARRAY_CSV_HELPER_TEXT);
+
+    cy.wait("@updateLayout");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:36]: Use `aghelepr.ValidateNetworkStatus`.
FYI: Changing method name to `AssertNetworkStatus` in Cy 12 PR

Comment (rajatagrawal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:36]: Done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
 const commonlocators = require("../../../../../locators/commonlocators.json");
 const dsl = require("../../../../../fixtures/filePickerTableDSL.json");
+import { agHelper, locators } from "../../../../../support/Objects/ObjectsCore";
 
 const widgetName = "filepickerwidgetv2";
 const ARRAY_CSV_HELPER_TEXT = `All non CSV, XLS(X), JSON or TSV filetypes will have an empty value`;
 const ObjectsRegistry =
   require("../../../../../support/Objects/Registry").ObjectsRegistry;
 let propPane = ObjectsRegistry.PropertyPane;
+let table = ObjectsRegistry.Table;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:10]: Take propPane & table also from ObjectsCore.

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:10]: `ObjectsRegistry` is not required.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 4]
```diff
@@ -121,27 +137,38 @@ describe("File picker widget v2", () => {
...
+      "baseUrl",
+    );
+    cy.get(commonlocators.filePickerRemoveButton).click({ force: true });
+
+    agHelper.ValidateNetworkStatus("@updateLayout");
+
+    // The text widget takes a bit of time to load the values
+    cy.wait(2000);
+
+    cy.get(locators._widgetInDeployed("textwidget")).should("have.text", "");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:162]: Use `agHelper.AssertText`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 5]
```diff
@@ -121,27 +137,38 @@ describe("File picker widget v2", () => {
...
       .selectFile("cypress/fixtures/testdata.json", { force: true });
-    cy.get(".t--widget-textwidget").should("contain", "baseUrl");
-    cy.get(".uppy-Dashboard-Item-action--remove").click({ force: true });
-    cy.wait(3000);
-    cy.get(".t--widget-textwidget").should("have.text", "");
+    cy.get(locators._widgetInDeployed("textwidget")).should(
+      "contain",
+      "baseUrl",
+    );
+    cy.get(commonlocators.filePickerRemoveButton).click({ force: true });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:155]: Use `agHelper.GetNClick`

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:155]: Here & all newly introduced places.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js [hunk 6]
```diff
@@ -121,27 +137,38 @@ describe("File picker widget v2", () => {
...
 
     cy.selectDropdownValue(commonlocators.filePickerDataFormat, "Binary");
     cy.get(commonlocators.filePickerInput)
       .first()
       .selectFile("cypress/fixtures/testdata.json", { force: true });
-    cy.get(".t--widget-textwidget").should("contain", "baseUrl");
-    cy.get(".uppy-Dashboard-Item-action--remove").click({ force: true });
+    cy.get(locators._widgetInDeployed("textwidget")).should(
+      "contain",
+      "baseUrl",
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:170]: Use `agHelper.GetNAssertContains`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js (hunk 3 above)
Comment (rajatagrawal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:10]: Done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js (hunk 6 above)
Comment (rajatagrawal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:170]: Done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js (hunk 5 above)
Comment (rajatagrawal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:155]: Done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js (hunk 4 above)
Comment (rajatagrawal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Filepicker/FilePickerV2_CSV_spec.js:162]: Done

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress changes!

**Pull request merged**